### PR TITLE
Add integration tests for StatefulSetStartOrdinal feature

### PR DESCRIPTION
**Pull request by pwschuurman**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Add integration test for KEP-3335:
 * https://github.com/kubernetes/enhancements/issues/3335

#### Special notes for your reviewer:
Adding integration test for Beta promotion in 1.27

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: test/integration/statefulset/statefulset_test.go [hunk 1]
```diff
@@ -479,3 +481,90 @@ func TestAutodeleteOwnerRefs(t *testing.T) {
...
+				Start: 2,
+			},
+			replicas:         7,
+			expectedPodNames: []string{"sts-2", "sts-3", "sts-4", "sts-5", "sts-6", "sts-7", "sts-8"},
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.StatefulSetStartOrdinal, true)()
+			closeFn, rm, informers, c := scSetup(t)
```
Comment (aojea) [inline test/integration/statefulset/statefulset_test.go:523]: can you check how long it takes to run these tests, and if is considerable to see if we can reuse the same setup for all the tests using namespaces for avoid test pollution?

Comment (pwschuurman) [reply test/integration/statefulset/statefulset_test.go:523]: Thanks @aojea, what's the best way to validate the run-time of a set of tests?

Comment (pwschuurman) [reply test/integration/statefulset/statefulset_test.go]: Thanks, was able to isolate these tests with:

```
go test ./test/integration/statefulset -v -run 'TestStatefulSetStartOrdinal'
```

This reduces the test run time time from 20s -> ~5s

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/triage accepted
/priority backlog

**Pull request merged**